src/dmd/dsymbol.d

```diff
@@ -305,24 +305,24 @@ extern (C++) class Dsymbol : RootObject
         va_end(ap);
     }
 
-    final void checkDeprecated(Loc loc, Scope* sc)
+    final bool checkDeprecated(Loc loc, Scope* sc)
     {
         if (global.params.useDeprecated != 1 && isDeprecated())
         {
             // Don't complain if we're inside a deprecated symbol's scope
             for (Dsymbol sp = sc.parent; sp; sp = sp.parent)
             {
                 if (sp.isDeprecated())
-                    return;
+                    return true;
             }
             for (Scope* sc2 = sc; sc2; sc2 = sc2.enclosing)
             {
                 if (sc2.scopesym && sc2.scopesym.isDeprecated())
-                    return;
+                    return true;
 
                 // If inside a StorageClassDeclaration that is deprecated
                 if (sc2.stc & STC.deprecated_)
-                    return;
+                    return true;
```

Comment:
Not sure about this. The definition for the other symbol is that `true` is returned if errors occur. In these cases error don't occur because it's allowed to e.g. use deprecated from deprecated.

Reply:
Gah! you're right.

Reply:
Fixed.  Thanks.